tools/devops/automation/templates/tests/run-tests.yml

```
@@ -250,6 +250,8 @@ steps:
   displayName: 'Run tests'
   name: runTests # not to be confused with the displayName, this is used to later use the name of the step to access the output variables from an other job
   timeoutInMinutes: 720
+  env:
+    BUILD_REVISION: $(Build.SourceVersion)
```

This is not correct AFAIK. Remember that we have the refs/PRID/merge when we checkout, but we are doing a git checkout regs/PRID/head (so that we build the change in the PR not the merge of main and the PR). That step changes the hash of the code. $(Build.SourceVersion) on a PR will point to the commit of refs/PRID/merge which is not the correct one. We probably want to use the env variable here: https://github.com/xamarin/xamarin-macios/blob/main/tools/devops/automation/templates/common/checkout.yml#L18
```suggestion
    BUILD_REVISION: $(GIT_HASH)
```
The logic in the checkout template will set the GIT_HASH variable to:
* the current change in a CI build.
* the change id of the refs/PRID/head

---

`GIT_HASH` isn't set later when running the tests, so is it set here?

The exact value doesn't really matter, as long as it's not empty (so I could set it to something like `BUILD_REVISION: "SomeHashOnCI"`)

---

It is @rolfbjarne  I decided that it was better to have all jobs in the same hash, so this was added to this template: https://github.com/xamarin/xamarin-macios/blob/main/tools/devops/automation/templates/common/checkout.yml which is executed here: https://github.com/xamarin/xamarin-macios/blob/main/tools/devops/automation/templates/tests/build.yml#L52 so it should be there, if not, is my bug.

---

but if it just needs not to be empty, I'm fine with it.